### PR TITLE
Refactor Layer component to use hooks

### PR DESCRIPTION
**Pull request by husseyexplores**
#### What does this PR do?
This PR refactors Layer component to use hooks.

#### Where should the reviewer start?
`/src/js/components/Layer/Layer.js`
`/src/js/components/Layer/LayerContainer.js`

#### What testing has been done on this PR?
Storybook

#### How should this be manually tested?
Storybook

#### Any background context you want to provide?
Test `should be null prior to mounting, displayed after mount` is failing because it uses implementation details to test the component. Since `Layer` is now a functional component, we can no longer use `ref` to access the instance and update state manually in our test or call `useEffect`. Any ideas about how to make that work or is this test even a valid test now?
[(This is the test I'm referring to)](https://github.com/grommet/grommet/blob/master/src/js/components/Layer/__tests__/Layer-test.js#L215)

#### What are the relevant issues?
#3394 

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

**Comment by ShimiSun** [comment]
I think you need to fix the failing test, could you please take a look?
```
Summary of all failing tests
 FAIL  src/js/components/Layer/__tests__/Layer-test.js
  ● Layer › should be null prior to mounting, displayed after mount

    TypeError: Cannot read property 'setState' of null

      213 |     );
      214 | 
    > 215 |     ref.current.setState({ islayerContainerAvailable: false });
          |                 ^
      216 |     expect(queryByTestId(document, 'test-layer-container')).toBeNull();
      217 | 
      218 |     ref.current.componentDidMount();

      at Object.setState (src/js/components/Layer/__tests__/Layer-test.js:215:17)


Test Suites: 1 failed, 51 passed, 52 total
Tests:       1 failed, 446 passed, 447 total
Snapshots:   556 passed, 556 total
Time:        37.295s
```

File: src/js/components/Layer/Layer.js [hunk 1]
```diff
@@ -1,76 +1,78 @@
...
-      : null;
-  }
-}
+Layer.defaultProps = {
+  full: false,
+  margin: 'none',
+  modal: true,
+  position: 'center',
+  responsive: true,
+};
```
Comment (ShimiSun) [inline src/js/components/Layer/Layer.js]: Can we please move the default props assignment to be part of the functional component declaration (line #9)?

File: src/js/components/Layer/LayerContainer.js [hunk 2]
```diff
@@ -17,137 +17,120 @@ const HiddenAnchor = styled.a`
...
+  }
+  return content;
+};
+
+LayerContainer.defaultProps = {
+  full: false,
+  margin: 'none',
+  modal: true,
+  position: 'center',
+};
```
Comment (ShimiSun) [inline src/js/components/Layer/LayerContainer.js]: Same comment here to move the assignment to be part of the declaration.

**Comment by husseyexplores** [comment]
> #### Any background context you want to provide?
> Test `should be null prior to mounting, displayed after mount` is failing because it uses implementation details to test the component. Since `Layer` is now a functional component, we can no longer use `ref` to access the instance and update state manually in our test or call `useEffect`. Any ideas about how to make that work or is this test even a valid test now?

File: src/js/components/Layer/Layer.js (hunk 1 above)
Comment (husseyexplores) [reply src/js/components/Layer/Layer.js]: If we define all the `defaultProps` on line #9, in the function parameters, then we would also need to pass every single prop into the `LayerContainer` component manually. (line #65)
That feels a bit awkward to me.
This is the only reason I didn't destructure the two props needed on line #9 but did it on line #10 so we could just pass along the props as-is to the `LayerContainer` component.

If being explicit is a requirement, I have no problem doing it the way you recommend. I just expressed my reasoning.

File: src/js/components/Layer/LayerContainer.js (hunk 2 above)
Comment (husseyexplores) [reply src/js/components/Layer/LayerContainer.js]: Made the change.

**Comment by ShimiSun** [comment]
I need to find time to dive into this, unfortunately, I don't have a straight answer.

**Comment by ShimiSun** [comment]
Thanks for contributing! I'll be closing this PR since refactoring Layer to use hooks was taken care of in other efforts.

**Pull request closed without merging**